src/config.cc

```diff
@@ -0,0 +1,8 @@
+#include "config.h"
+#include <mutex>
+namespace valhalla {
+const boost::property_tree::ptree& config(const std::string& config_file_or_inline) {
+  static config_singleton_t instance(config_file_or_inline);
+  return instance.config_;
+}
+}; // namespace valhalla
```

Comment:
why break this out into a .cc file? it's sufficiently small for a header function no? also `mutex` shouldn't be needed as include?